alf/environments/suite_dmlab.py

```diff
@@ -0,0 +1,195 @@
+# Copyright (c) 2019 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import gym
+import gin.tf
+from tf_agents.environments import suite_gym
+from tf_agents.environments import wrappers
+from alf.environments.suite_socialbot import ProcessPyEnvironment
+
+# `DeepmindLab` are required,
+#   see `https://github.com/deepmind/lab` to build `DeepmindLab`
+
+try:
+    import deepmind_lab
+except ImportError:
+    deepmind_lab = None
+
+
+def is_available():
+    return deepmind_lab is not None
+
+
+@gin.configurable
+def action_discretize(action_spec,
+                      look_left_right_pixels_per_frame=(-20, 20),
+                      look_down_up_pixels_per_frame=(-10, 10),
+                      strafe_left_right=(-1, 1),
+                      move_back_forward=(-1, 1),
+                      fire=(),
+                      jump=(1, ),
+                      crouch=(1, ),
+                      **kwargs):
+    """Discretize action from action_spec
```

Comment:
More details: describe what the discretization does by giving examples, especially what we should expect about the output representation.